src/core/Target_location.ml

```diff
@@ -0,0 +1,82 @@
+(* Cooper Pierce
+ *
+ * Copyright (C) 2023 Semgrep Inc.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * version 2.1 as published by the Free Software Foundation, with the
+ * special exception on linking described in file LICENSE.
+ *
+ * This library is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the file
+ * LICENSE for more details.
+ *)
+
+type t = Code of code | Lockfile of lockfile [@@deriving show]
+
+(* The location of a "normal" semgrep target, comprising source code (or for
+ * regex/generic, arbitrary text data) to be executed.
+ *)
+and code = {
```

Comment:
It is called "code" not really to do anything with the code product, but rather because it is source code. I want to avoid naming this `source` for obvious reasons.